### DISCUSSION TITLE
Find reason for a revision?

### DISCUSSION THREAD

**Discussion by ja-gooding** [Homebrew/discussions#2626] [Tap maintenance and Homebrew development] [answered]
Is there a way to check and know what caused a revision to a formula?

**Comment by Bo98** [comment, marked as answer]
Taps are git repos so just have a look at the commit history.

e.g.

`git -C "$(brew --repo homebrew/core)" log Formula/gcc.rb`

Not all commits necessarily result in a revision bump, but you can see what each commit did in detail by finding the commit hash in the log and doing `git show <commit>`.

You can mostly ignore the "update bottle" commits. Though if you see a version/revision change in those bottle update commit messages, then the commit _prior_ is likely the one that did the version or revision bump.

**Reply by ja-gooding** [reply]
Thank you, I wasn't looking back enough, you're right the _prior_ one is what caused the revision bump in this case. Thanks.
